src/main/java/io/github/cottonmc/energy/impl/EnergyHandler.java

```diff
@@ -0,0 +1,63 @@
+package io.github.cottonmc.energy.impl;
+
+import io.github.cottonmc.energy.api.ActionType;
+import io.github.cottonmc.energy.api.EnergyComponent;
+
+import javax.annotation.Nonnull;
+
+public class EnergyHandler implements EnergyComponent {
+
+	private int maxEnergy;
+	private int currentEnergy = 0;
+
+	public EnergyHandler(int maxEnergy) {
+		this.maxEnergy = maxEnergy;
+	}
+
+	@Override
+	public int getMaxEnergy() {
+		return maxEnergy;
+	}
+
+	@Override
+	public int getCurrentEnergy() {
+		return currentEnergy;
+	}
+
+	@Override
+	public boolean canInsertEnergy() {
```

Comment:
I wonder if we shouldn't just consider canInsert / canExtract to be true unless specifically denied? We can always find out if there's room by asking getCurrent and getMax, or by simulating an insert.